src/sensors/humidity.rs

```diff
@@ -0,0 +1,44 @@
+//! Module providing humidity sensor functionality.
+
+use super::{LocalisedSensorMetadata, SensorMetadata, SensorTemplate, Sensors};
+use serde::{Deserialize, Serialize};
+use std::convert::TryInto;
+
+#[derive(Serialize, Deserialize, Default, Debug, Clone, PartialEq)]
+pub struct HumiditySensor {
+    #[serde(flatten)]
+    pub metadata: LocalisedSensorMetadata,
+    pub unit: String,
+    pub value: f64,
+}
+
+#[derive(Debug, Clone)]
+pub struct HumiditySensorTemplate {
+    pub metadata: SensorMetadata,
+    pub unit: String,
+}
+
+impl TryInto<HumiditySensor> for HumiditySensorTemplate {
+    type Error = Box<dyn std::error::Error>;
+
+    fn try_into(self) -> Result<HumiditySensor, Self::Error> {
+        Ok(HumiditySensor {
+            metadata: self.metadata.try_into()?,
```

Comment:
When the template has the `LocalisedSensorMetadata` this shouldn't be able to fail. So we could just implement `From<HumiditySensorTemplate> for HumiditySensor`.